flink-scala/src/main/scala/org/apache/flink/api/scala/GroupedDataSet.scala

```diff
@@ -282,27 +283,57 @@ class GroupedDataSet[T: ClassTag](
   }
 
   /**
-   * Creates a new [[DataSet]] by merging the elements of each group (elements with the same key)
-   * using an associative reduce function.
-   */
+    * Creates a new [[DataSet]] by merging the elements of each group (elements with the same key)
+    * using an associative reduce function.
+    */
   def reduce(fun: (T, T) => T): DataSet[T] = {
+    reduce(getCallLocationName(), fun, CombineHint.OPTIMIZER_CHOOSES)
+  }
+
+  /**
+   * Special [[reduce]] operation for explicitly telling the system what strategy to use for the
+   * combine phase.
+   * If null is given as the strategy, then the optimizer will pick the strategy.
+   */
+  def reduce(fun: (T, T) => T, strategy: CombineHint): DataSet[T] = {
```

Comment:
There is a problem with this in the Scala API: the `wrap()` calls hide the `ReduceOperator` from the user.

Would it be OK if I added optional arguments instead of the overloads like this?
```
def reduce(fun: (T, T) => T, strategy: CombineHint = CombineHint.OPTIMIZER_CHOOSES): DataSet[T] = {
```